example/util/getPoolBidDetails.tsx

```diff
@@ -0,0 +1,154 @@
+import { PublicKey, Connection, Keypair } from "@solana/web3.js";
+import { AnchorProvider, BN } from "@project-serum/anchor";
+import {
+  TensorSwapSDK,
+  TakerSide,
+  CurveType,
+  PoolType,
+  castPoolConfigAnchor,
+  computeMakerAmountCount,
+} from "@tensor-oss/tensorswap-sdk";
+import { conn, keypair } from "./common";
+import Big from "big.js";
+import { SimpleWallet } from "./model/SimpleWallet";
+
+const provider = new AnchorProvider(conn,  new SimpleWallet(keypair), {
+  commitment: "confirmed",
+});
+const swapSdk = new TensorSwapSDK({ provider });
+const HUNDRED_PCT_BPS = 100_00;
+
+async function getPoolBidDetails(poolAddress: any) {
+  // fetch pool
+  const pool = await swapSdk.fetchPool(new PublicKey(poolAddress));
+  const config = castPoolConfigAnchor(pool.config);
+
+  // return early if pool is one-sided listing-only
+  if (config.poolType == PoolType.NFT) {
+    console.log("Listing-only pool can not bid.");
+    return null;
+  }
+
+  // fetch balance of margin account if pool is attached to escrow, fetch balance of solEscrow instead if pool is not attached to escrow
+  const solBalanceLamports =
+    pool.margin != null
+      ? new BN(await conn.getBalance(pool.margin))
+      : new BN(await conn.getBalance(pool.solEscrow));
+
+  // retrieve amount of possible bids, total lamports needed for that amount of bids and initial price of the pool
+  const { allowedCount, totalAmount, initialPrice } = computeMakerAmountCount({
+    desired: { total: solBalanceLamports },
+    maxCountWhenInfinite: 1000,
+    takerSide: TakerSide.Sell,
+    extraNFTsSelected: 0,
+    config,
+    takerSellCount: pool.takerSellCount,
+    takerBuyCount: pool.takerBuyCount,
+    maxTakerSellCount: pool.maxTakerSellCount,
+    statsTakerSellCount: pool.stats.takerSellCount,
+    statsTakerBuyCount: pool.stats.takerBuyCount,
+    marginated: pool.margin !== null,
+  });
+
+  // return early if amount of possible bids is 0
+  if (allowedCount == 0) {
+    console.log(
+      "Pool is out of funds or reached maxTakerSellCount and won't bid anymore."
+    );
+    return null;
+  }
+
+  // retrieve initial highest bid price if pool is double sided (config.startingPrice would be the initial lowest list price in that case)
+  var startingPriceBidSide: Big;
```

Comment:
In most cases, prefer let to var unless the variable needs to be declared globally." This provides more context and suggests let as the default choice.